src/components/Card/Card.tsx

```diff
@@ -45,8 +48,55 @@ function Card({
     };
   }, [cardholderRef]);
 
+  const [springs, api] = useSpring(
+    () => ({
+      x: 0,
+      y: 0,
+      config: {
+        mass: 2,
+        friction: 14,
+        tension: 120,
+      },
+    }),
+    []
+  );
+
+  const handleMouseLeave = () => {
+    const regex = /translate\(([-\d.]+)px,\s*([-\d.]+)px\)/;
```

Comment:
minor thing, but still: `regex` is used inside `if` statement only, so it might be better to define it there. You could also name it less generically: something like `translateRegex`

Reply:
You could also add named groups before each searched value for more idiomatic code:
```ts
const regex =
  // same thing, just added ?<offsetX> and ?<offsetY> before each group
  /translate\((?<offsetX>[-\d.]+)px,\s*(?<offsetY>[-\d.]+)px\)/;

const transformValue = cardholderRef.current.style.transform;

const matches = regex.exec(transformValue);

if (matches?.groups?.length) {
  const { offsetX, offsetY } = matches.groups;

  springRef.start({
          from: {
            x: parseFloat(offsetX),
            y: parseFloat(offsetY),
```

Reply:
Wow, named capture groups blew my mind! Implemented this comment with one small change: matches.groups returns an object, not an array, so a length check doesn't apply here

Reply:
right, my bad